samcli/commands/local/lib/function_url_handler.py

```diff
@@ -0,0 +1,411 @@
+        else:
+            # Default permissive CORS for local development
+            headers["Access-Control-Allow-Origin"] = "*"
+            headers["Access-Control-Allow-Methods"] = "GET, POST, PUT, DELETE, PATCH, HEAD, OPTIONS"
+            headers["Access-Control-Allow-Headers"] = "*"
+            headers["Access-Control-Max-Age"] = "86400"
+
+        return Response("", status=200, headers=headers)
+
+    def _get_cors_headers(self):
+        """Get CORS headers based on configuration"""
+        cors_config = self.function_config.get("cors", {})
+
+        if not cors_config:
+            return {}
+
+        headers = {}
+
+        origins = cors_config.get("AllowOrigins", ["*"])
+        headers["Access-Control-Allow-Origin"] = ", ".join(origins)
+
+        if cors_config.get("AllowCredentials"):
+            headers["Access-Control-Allow-Credentials"] = "true"
+
+        expose_headers = cors_config.get("ExposeHeaders")
+        if expose_headers:
+            headers["Access-Control-Expose-Headers"] = ", ".join(expose_headers)
+
+        return headers
+
+    def _validate_iam_auth(self, request) -> bool:
+        """
+        Validate IAM authorization (simplified for local testing)
+
+        In production, this would validate AWS SigV4 signatures.
+        For local development, we just check for the presence of an Authorization header.
+
+        WARNING: This is a mock implementation for local testing only.
+        Real IAM authorization with signature validation is not performed.
+        """
+        if self.disable_authorizer:
+            return True
+
+        # Simple check for Authorization header presence
+        auth_header = request.headers.get("Authorization")
+        if not auth_header:
+            LOG.debug("No Authorization header found")
+            return False
+
+        # In local mode, accept any Authorization header that starts with "AWS4-HMAC-SHA256"
+        if auth_header.startswith("AWS4-HMAC-SHA256"):
+            LOG.warning(
+                "IAM authorization is simplified for local testing. "
+                "Any AWS4-HMAC-SHA256 header is accepted. "
+                "Use --disable-authorizer flag to skip authorization checks entirely."
+            )
+            return True
+
+        LOG.debug(f"Invalid Authorization header format: {auth_header[:20]}...")
+        return False
+
+    def start(self):
+        """Start the Function URL service"""
+        LOG.info(f"Starting Function URL for {self.function_name} at " f"http://{self.host}:{self.port}/")
+
+        # Run Flask app in a separate thread
+        self._server_thread = Thread(target=self._run_flask, daemon=True)
+        self._server_thread.start()
+
+    def _run_flask(self):
+        """Run the Flask application"""
+        try:
+            self.app.run(
+                host=self.host, port=self.port, threaded=True, use_reloader=False, use_debugger=False, debug=False
```

Comment:
Why are you extending from `BaseLocalService` class if you're not using any of the Flask functionality from there?